pkg/query-service/app/logs/v3/query_builder.go

```diff
@@ -273,82 +291,56 @@ func buildLogsQuery(panelType v3.PanelType, start, end, step int64, mq *v3.Build
 
 // groupBy returns a string of comma separated tags for group by clause
 // `ts` is always added to the group by clause
-func groupBy(panelType v3.PanelType, tags ...string) string {
-	if panelType == v3.PanelTypeGraph || panelType == v3.PanelTypeValue {
+func groupBy(panelType v3.PanelType, graphLimitQtype string, tags ...string) string {
```

Comment:
If it helps, you can have separate functions just for this first query kind.

Reply:
Yeah, anyway I have to clean it up in the refractoring processs. Keeping it as it is for now.

Reply:
Sure